### ISSUE TITLE
Add preference to disable reporting the currently played video to Now Playing

### ISSUE THREAD

**Issue by unikitty37** [iina/iina#3896]
**What you want IINA to do:**

I would like IINA to have a preference to not report what it is playing to the Now Playing service.

**What IINA does currently:**

IINA always reports what it is playing to the Now Playing service.

**Why you think this should be added:**

IINA allows the user to disable the Open Recent menu, presumably as a privacy feature.

However, the Now Playing service also keeps a list of items that has been played, and provides this as part of its menu bar item. Even if the menu bar item is disabled in System Preferences, re-enabling it will show the last item played in IINA.

Deleting the video and emptying the Trash still does not remove the information from Now Playing, though it does (obviously!) prevent it from being re-played if the Play button is clicked in Now Playing.

If I watch a video where the filename features confidential information (name of project, for example), delete the video, and empty the Trash, it's reasonable to assume that somebody accessing the machine later will not be able to see what that video was. Now Playing breaks this, unless I remember to play something with an innocuous name like "kittens.mp4" in IINA afterwards.

If the user does not want IINA to have a list of recently-played videos in IINA's Open Recent, it's quite likely that they won't want the last one to appear in the system Now Playing menu bar item until they play something else. There should be an option to disable reporting to Now Playing.

**Examples of other projects that have something similar:**

I can't think of any that have an option; there are probably some that don't report to Now Playing *at all*, but I can't find any. In any case, removing this feature is likely to be unpopular with many other people; a preference would be the best option.

**Comment by low-batt** [comment]
Thank you for the nice writeup explaining in detail the reason this preference should be added.

**Comment by low-batt** [comment]
Try this:
- Open IINA
- Select `Preferences...`
- Click on `Key Bindings`
- Uncheck the preference `Use system media control`
- **Restart** IINA

IINA will no longer report the video being played to macOS Now Playing. Unfortunately this also means the media keys will not be able to control IINA.

I'm treating this issue is a request for finer grained control. Allow use of media keys, but don't report private information.

**Comment by saagarjha** [comment]
Perhaps we should check for the file disappearing out from under the app and closing it?

**Comment by unikitty37** [comment]
> IINA will no longer report the video being played to macOS Now Playing. Unfortunately this also means the media keys will not be able to control IINA.

Thanks! That's actually fine for me, as I don't usually use the system media keys in any case :)

But finer grained control would certainly be useful for people who do, so I'll leave this open.

**Comment by low-batt** [comment]
Yes, keep open. This would be one part of a "Private Playback" feature, similar to how web browsers offer "Private Browsing". This is related to issue #3515 which is asking for no record to be kept for privacy reasons.
